odoo/tests/common.py

```diff
@@ -594,6 +601,385 @@ def phantom_js(self, url_path, code, ready="window", login=None, timeout=60, **k
+                if r.ok:
+                    return r.json()
+                return {'status_code': r.status_code}
+            except requests.ConnectionError:
+                time.sleep(1)
+                timeout -= 1
+        _logger.error('Could not connect to chrome debugger')
+        raise unittest.SkipTest("Cannot connect to chrome headless")
+
+    def _open_websocket(self):
+        self.ws = websocket.create_connection(self.ws_url)
+        if self.ws.getstatus() != 101:
+            raise unittest.SkipTest("Cannot connect to chrome dev tools")
+        self.ws.settimeout(0.01)
+
+    def _websocket_send(self, method, params=None):
+        """
+        send chrome devtools protocol commands through websocket
+        """
+        sent_id = self.request_id
+        payload = {
+            'method' : method,
+            'id' :  sent_id,
+        }
+        if params:
+            payload.update({'params' : params})
+        self.ws.send(json.dumps(payload))
+        self.request_id += 1
+        return sent_id
+
+    def _websocket_wait_id(self, awaited_id, timeout=10):
+        """
+        blocking wait for a certain id in a response
+        warning other messages are discarded
+        """
+        start_time = time.time()
+        while time.time() - start_time < timeout:
+            try:
+                res = json.loads(self.ws.recv())
+            except websocket.WebSocketTimeoutException:
+                res = None
+            if res and res.get('id') == awaited_id:
+                return res
+        _logger.info('timeout exceeded while waiting for id : %d' % awaited_id)
+        return {}
+
+    def _websocket_wait_event(self, method, params=None, timeout=10):
+        """
+        blocking wait for a particular event method and eventually a dict of params
+        """
+        start_time = time.time()
+        while time.time() - start_time < timeout:
+            try:
+                res = json.loads(self.ws.recv())
+            except websocket.WebSocketTimeoutException:
+                res = None
+            if res and res.get('method', '') == method:
+                if params:
+                    if set(params).issubset(set(res.get('params', {}))):
+                        return res
+                else:
+                    return res
+            elif res:
+                _logger.debug('chrome devtools protocol event: %s' % res)
+        _logger.info('timeout exceeded while waiting for : %s' % method)
+
+
+    def take_screenshot(self, prefix='failed'):
+        ss_id = self._websocket_send('Page.captureScreenshot')
+        _logger.info('Asked for screenshot (id: %s)' % ss_id)
+        res = self._websocket_wait_id(ss_id)
+        base_png = res.get('result', {}).get('data')
+        decoded = base64.decodebytes(bytes(base_png.encode('utf-8')))
+        timestamp = datetime.now().strftime('%Y%m%d_%H%M%S_%f')
+        outpath = os.path.join(self.screenshots_dir, '%s_%s.png' % (prefix,timestamp))
```

Comment:
'%s_failed_%Y%m%d_%H%M%S_%f_%d' % (os.path.splitext('logfile.log')[0], 1)

Reply:
On the runbot, the stderr and stdout are redirected to a file, there is no --logfile argument

https://github.com/odoo/runbot/blob/e815571d14163c8af6c19637c244ff022abaa883/runbot/models/build.py#L716-L718